code/modules/mod/modules/modules_supply.dm

```diff
@@ -0,0 +1,537 @@
+	var/angle = get_angle(mod.wearer, target)
+	mod.wearer.transform = mod.wearer.transform.Turn(angle)
+	mod.wearer.throw_at(get_ranged_target_turf_direct(mod.wearer, target, power), \
+		range = power, speed = max(round(0.2*power), 1), thrower = mod.wearer, spin = FALSE, \
+		callback = CALLBACK(src, PROC_REF(on_throw_end), mod.wearer, -angle))
+
+/obj/item/mod/module/hydraulic/proc/on_throw_end(mob/user, angle)
+	if(!user)
+		return
+	user.transform = user.transform.Turn(angle)
+	user.cut_overlay(charge_up_overlay)
+
+/obj/item/mod/module/magnet
+	name = "MOD loader hydraulic magnet module"
+	desc = "A powerful hydraulic electromagnet able to launch crates and lockers towards the user, and keep them attached."
+	icon_state = "magnet_loader"
+	module_type = MODULE_ACTIVE
+	removable = FALSE
+	use_power_cost = DEFAULT_CHARGE_DRAIN * 3
+	incompatible_modules = list(/obj/item/mod/module/magnet)
+	cooldown_time = 1.5 SECONDS
+	overlay_state_active = "module_magnet"
+	use_mod_colors = TRUE
+
+/obj/item/mod/module/magnet/on_select_use(atom/target)
+	. = ..()
+	if(!.)
+		return
+	if(istype(mod.wearer.pulling, /obj/structure/closet))
+		var/obj/structure/closet/locker = mod.wearer.pulling
+		playsound(locker, 'sound/effects/gravhit.ogg', 75, TRUE)
+		locker.forceMove(mod.wearer.loc)
+		locker.throw_at(target, range = 7, speed = 4, thrower = mod.wearer)
+		return
+	if(!istype(target, /obj/structure/closet) || !(target in view(mod.wearer)))
+		to_chat(mod.wearer, "<span class='warning'>Invalid target!</span>")
+		return
+	var/obj/structure/closet/locker = target
+	if(locker.anchored || locker.move_resist >= MOVE_FORCE_OVERPOWERING)
+		return
+	playsound(locker, 'sound/effects/gravhit.ogg', 75, TRUE)
+	locker.throw_at(get_step_towards(mod.wearer, target), range = 7, speed = 3, force = MOVE_FORCE_WEAK, \
+		callback = CALLBACK(src, PROC_REF(check_locker), locker))
+
+/obj/item/mod/module/magnet/on_deactivation(display_message = TRUE, deleting = FALSE)
+	. = ..()
+	if(!.)
+		return
+	if(istype(mod.wearer.pulling, /obj/structure/closet))
+		mod.wearer.stop_pulling()
+
+/obj/item/mod/module/magnet/proc/check_locker(obj/structure/closet/locker)
+	if(!mod?.wearer)
+		return
+	if(!locker.Adjacent(mod.wearer) || !isturf(locker.loc) || !isturf(mod.wearer.loc))
+		return
+	mod.wearer.start_pulling(locker)
+
+/obj/item/mod/module/ash_accretion
+	name = "MOD ash accretion module"
+	desc = "A module that collects ash from the terrain, covering the suit in a protective layer, this layer is \
+		lost when moving across standard terrain."
+	icon_state = "ash_accretion"
+	removable = FALSE
+	incompatible_modules = list(/obj/item/mod/module/ash_accretion)
+	overlay_state_inactive = "module_ash"
+	use_mod_colors = TRUE
+	/// How many tiles we can travel to max out the armor.
+	var/max_traveled_tiles = 10
+	/// How many tiles we traveled through.
+	var/traveled_tiles = 0
+	/// Armor values per tile.
+	var/list/armor_mod_1 = /obj/item/mod/armor/mod_ash_accretion
```

Comment:
Is this suppose to be a list, or something that's just assigned one value?

Reply:
one value, removed from being a list